packages/fuels-signers/src/wallet.rs

```diff
@@ -222,6 +225,46 @@ impl Wallet {
             .map_err(Into::into)
     }
 
+    pub async fn get_messages(&self) -> Result<Vec<schema::message::Message>, Error> {
+        Ok(self.get_provider()?.get_messages(&self.address).await?)
+    }
+
+    pub async fn get_inputs_for_messages(&self, witness_index: u8) -> Result<Vec<Input>, Error> {
+        let as_u8_slice = |v: &[i32]| -> &[u8] {
+            let element_size = mem::size_of::<i32>();
+            unsafe { from_raw_parts(v.as_ptr() as *const u8, v.len() * element_size) }
```

Comment:
Can we add a quick comment on the safety guarantees around this `unsafe` block? You should explain why it's safe or when it wouldn't be safe. 

Here's an example of this type of comment from the `fuel-vm`: https://github.com/FuelLabs/fuel-vm/blob/e823401c1f087d47ddab4ec863d34ea6c6c1b3af/src/interpreter/blockchain.rs#L249-L250

Reply:
I think Ahmed's solution is better, so I chose it. https://github.com/FuelLabs/fuels-rs/pull/555#discussion_r967521873 There is no need to use this unsafe method.